src/core/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,243 @@
+import React, { Component, forwardRef } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { SuomifiThemeProp, SuomifiThemeConsumer } from '../theme';
+import { baseStyles } from './Pagination.baseStyles';
+import { HtmlSpan, HtmlNav, HtmlDiv } from '../../reset';
+import { PageInput, PageInputValue } from './PageInput/PageInput';
+import { Button } from '../Button/Button';
+import { VisuallyHidden } from '../VisuallyHidden/VisuallyHidden';
+import { AutoId } from '../utils/AutoId/AutoId';
+
+export interface PageInputProps {
+  /** Page input action button label for screen readers  */
+  buttonText: string;
+  /** Input placeholder text */
+  inputPlaceholderText: string;
+  /** Error text shown with invalid page number input */
+  invalidValueErrorText: (value: PageInputValue) => string;
+  /** Visually hidden label for input field */
+  labelText: string;
+}
+
+type ShowInputProps =
+  | {
+      pageInput: false;
+      pageInputProps?: never;
+    }
+  | {
+      /** Show input field for page number */
+      pageInput?: true;
+      /** Props for page input field */
+      pageInputProps: PageInputProps;
```

Comment:
Can it be added to Styleguidist somehow, what this PageInputProps contains. For example <pre> tag with interface information in this comment?

Reply:
This is now added with <pre> tags